include/jemalloc/internal/rtree.h

```diff
@@ -235,17 +266,54 @@ rtree_leaf_elm_bits_decode(uintptr_t bits) {
 	    >> RTREE_NHIB) & low_bit_mask);
 #    endif
 	assert((uintptr_t)contents.edata % (uintptr_t)EDATA_ALIGNMENT == 0);
+#    ifdef LIMIT_USIZE_GAP
+	if (read_usize) {
+		rtree_read_metadata_usize(&contents);
+	} else {
+		contents.metadata.usize = 0;
+	}
+#    endif
 	return contents;
 }
 
 #  endif /* RTREE_LEAF_COMPACT */
 
+JEMALLOC_ALWAYS_INLINE rtree_contents_t
+rtree_leaf_elm_read_fast(tsdn_t *tsdn, rtree_t *rtree, rtree_leaf_elm_t *elm,
```

Comment:
This is only used in fast path. On the fast path, although `dependent` is true, we don't want to read the usize from the rtree because: 
1. if `usize != sz_index2size(ind)`, it should not be on the fast path (tcache cannot handle such cases), so there is no need for reading it.
2. reading the usize on the fast path is extremely expensive (adds ~20 instructions to the original 50 instructions). 

With this optimization, the assembly stays the same.